### PR TITLE
Server Responds with Hello World

### PR DESCRIPTION
**Pull request by gilroman**

### CODE REVIEW

File: src/main/java/gil/server/ArgumentsParser.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        throw new Exception();
+    }
+
+    public int getPortFlagValue(String[] args) throws Exception {
+        int portFlagIndex = getPortFlagIndexNumber(args);
+        int portFlagValueIndex = portFlagIndex + 1;
+        if (portIsValid(args[portFlagValueIndex])){
+            return Integer.parseInt(args[portFlagValueIndex]);
+        }
+        throw new Exception();
```
Comment (tbaik) [inline src/main/java/gil/server/ArgumentsParser.java]: Can you be more specific about this Exception? Either create a new exception or give it some description.

File: src/main/java/gil/server/Response.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package gil.server;
+
+import java.io.UnsupportedEncodingException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+public class Response {
+    private String startLine = "HTTP/1.1 200 OK";
+    private String date = "Date: " + java.time.format.DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC));
```
Comment (tbaik) [inline src/main/java/gil/server/Response.java]: can you import this at the top??

File: src/main/java/gil/server/ServerSocketWrapper.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package gil.server;
+
+import java.io.*;
```
Comment (tbaik) [inline src/main/java/gil/server/ServerSocketWrapper.java]: It's good practice to only import what you need. So please change all imports that do *

**Review by tbaik — CHANGES_REQUESTED** [review]

File: src/main/java/gil/server/ArgumentsParser.java (hunk 1 above)
Comment (gilroman) [reply src/main/java/gil/server/ArgumentsParser.java]: I added a MissingPortFlagException to address this.

File: src/main/java/gil/server/Response.java (hunk 2 above)
Comment (gilroman) [reply src/main/java/gil/server/Response.java]: Absolutely

File: src/main/java/gil/server/ServerSocketWrapper.java (hunk 3 above)
Comment (gilroman) [reply src/main/java/gil/server/ServerSocketWrapper.java]: Thanks for catching this.

File: src/test/java/gil/server/ServerTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package gil.server;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import org.junit.Test;
+import static org.junit.Assert.*;
```
Comment (tbaik) [inline src/test/java/gil/server/ServerTest.java]: missed one

Comment (gilroman) [reply src/test/java/gil/server/ServerTest.java]: Got it. It's fixed now.

**Pull request merged**